src/consumers.ts

```diff
@@ -129,31 +150,44 @@ class SecurityEngine extends AnalysisConsumer implements DiagnosticProducer
 {
     constructor(public context: IDependency, config: any) {
         super(config);
-        this.binding = {path: ['result', 'recommendation', 'component-analyses', 'cve']};
+        this.binding = {path: ['component_analyses', 'vulnerability']};
         /* recommendation to use a different version */
-        this.changeToBinding = {path: ['result', 'recommendation', 'change_to']};
+        this.changeToBinding = {path: ['recommended_versions']};
+        /* snyk registration link */
+        this.registrationLinkBinding = {path: ['registration_link']};
+        /* Diagnostic message */
+        this.messageBinding = {path: ['message']};
+        /* Publicly known Security Vulnerability count */
+        this.vulnerabilityCountBinding = {path: ['known_security_vulnerability_count']};
+        /* Private Security Advisory count */
+        this.advisoryCountBinding = {path: ['security_advisory_count']};
     }
 
     produce(ctx: any): Diagnostic[] {
         if (this.item.length > 0) {
-            let cveList = [];
-            for (let cve of this.item) {
-                cveList.push(cve['id'])
+            /* The diagnostic's severity. */
+            let diagSeverity;
+
+            if (this.vulnerabilityCount == 0 && this.advisoryCount > 0) {
+                diagSeverity = DiagnosticSeverity.Information; 
+            } else {
+                diagSeverity = DiagnosticSeverity.Error;
             }
-            let cves = cveList.join(' ');
 
             let diagnostic = {
-                severity: DiagnosticSeverity.Error,
+                severity: diagSeverity,
                 range: get_range(this.context.version),
-                message: `Application dependency ${this.context.name.value}-${this.context.version.value} is vulnerable: ${cves}`,
-                source: 'Dependency Analytics'
+                message: `${this.message}`,
+                source: 'Dependency Analytics',
+                code: `Find out more: ${this.registrationLink}` 
             };
 
             // TODO: this can be done lazily
-            if (this.changeTo != null) {
+            if (this.changeTo && (this.vulnerabilityCount > 0)) {
                 let codeAction: CodeAction = {
                     title: "Switch to recommended version " + this.changeTo,
                     diagnostics: [diagnostic],
+                    kind: CodeActionKind.QuickFix,  //provide a QuickFix option if recommended version is available
```

Comment:
Add a space after //
```suggestion
                    kind: CodeActionKind.QuickFix,  // Provide a QuickFix option if recommended version is available
```